src/storage/leveldb/store.ts

```diff
@@ -21,13 +20,9 @@ export default class LevelDBStore implements IStore {
   walletIndex: LevelWalletIndex;
   tokenIndex: LevelTokenIndex;
   dbpath: string;
-  xpubkey: string;
 
-  constructor(dbroot: string, xpubkey: string) {
-    // The xpubkey in the account or change path?
-    this.xpubkey = xpubkey;
-    const xpub = HDPublicKey.fromString(xpubkey);
-    const dbpath = path.join(dbroot, xpub.publicKey.toString());
+  constructor(dirpath: string, dbroot: string = 'hathor.data') {
+    const dbpath = path.join(dbroot, dirpath);
```

Comment:
If we add characters that are not accepted by the filesystem, will be error thrown here be easy to understand?

If the error message is too vague or misleading, I would suggest adding a validation and/or a `snakeCase(deburr(input))` kind of operation here ( reference [link](https://lodash.com/docs/4.17.15#deburr) )

Reply:
This is a good idea, but I think we should have this on a separate PR since it could also be used in other places.